body.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+)
+
+type body struct {
+	File   *os.File
+	Reader io.Reader
+	Offset int64
+	Size   int64
+}
+
+// Create a body.  A body is an io.Reader instance which reads from the file at
+// filename, starting at the `offset`th byte and reading up to `size` bytes in
+// total.
+func newBody(filename string, offset, size int64) (*body, error) {
+	if size == 0 {
+		return nil, errors.New("Cannot specify a size of 0")
+	}
+
+	file, err := os.Open(filename)
+
+	if err != nil {
+		return nil, err
+	}
+
+	b := body{file, nil, offset, size}
+
+	b.Reset()
```

Comment:
This could return an error if the file is a directory, so you should check for an error here.